src/main/scala/scalaz/analytics/AnalyticsModule.scala

```diff
@@ -78,16 +97,16 @@ trait AnalyticsModule {
   }
 
   /**
-   * The Operations supported by the core DataStream abstraction in scalaz-analytics.
+   * The dataset/datastream operations of scalaz-analytics
    */
-  trait SetOperations {
-    def union[A](l: DataStream[A], r: DataStream[A]): DataStream[A]
-    def intersect[A](l: DataStream[A], r: DataStream[A]): DataStream[A]
-    def except[A](l: DataStream[A], r: DataStream[A]): DataStream[A]
-    def distinct[A](d: DataStream[A]): DataStream[A]
-    def map[A, B](d: DataStream[A])(f: A =>: B): DataStream[B]
-    def sort[A](d: DataStream[A]): DataStream[A]
-    def distinctBy[A, B](d: DataStream[A])(by: A =>: B): DataStream[A]
+  trait Ops[F[_]] {
+    // Unbounded
+    def map[A, B](ds: F[A])(f: A =>: B): F[B]
+    def filter[A, B](ds: F[A])(f: A =>: Boolean): F[A]
+
+    // Bounded
+    def fold[A, B](ds: F[A])(window: Window)(initial: A =>: B)(f: (B, A) =>: B): F[B]
+    def distinct[A](ds: F[A])(window: Window): F[A]
```

Comment:
Very much tangential to the purpose of this PR but worth discussing anyway - is distinct a specialisation of `groupByAgg`? (where `groupByAgg` is a groupBy + aggregation (exposed as a single step as discussed in #4)).
If so - does it perhaps make sense to implement it in terms of this more foundational operation?